src/components/ui/TextArea.tsx

```diff
@@ -0,0 +1,108 @@
+import { ChangeEvent } from 'preact/compat'
+import { ErrorText, TextareaText } from 'components/ui/Text'
+import {
+  alignItems,
+  backgroundColor,
+  borderColor,
+  borderRadius,
+  borderWidth,
+  classnames,
+  display,
+  flexDirection,
+  flexGrow,
+  justifyContent,
+  outlineColor,
+  outlineStyle,
+  padding,
+  resize,
+  space,
+  transitionProperty,
+  width,
+} from 'classnames/tailwind'
+import { parseErrorText } from '@big-whale-labs/frontend-utils'
+import SuffixBlock from 'components/ui/SuffixBlock'
+import TextareaAutosize, {
+  TextareaAutosizeProps,
+} from 'react-textarea-autosize'
+
+const containerWithFooter = classnames(
+  display('flex'),
+  flexDirection('flex-col', 'md:flex-row'),
+  alignItems('items-stretch', 'md:items-start'),
+  space('space-y-2', 'md:space-x-2', 'md:space-y-0')
+)
+const innerWrapper = (isValid?: boolean) =>
+  classnames(
+    display('flex'),
+    flexDirection('flex-col'),
+    flexGrow('grow'),
+    alignItems('items-stretch'),
+    justifyContent('justify-between'),
+    borderWidth('border'),
+    borderColor(
+      isValid ? 'border-formal-accent-dimmed' : 'border-error',
+      'focus-within:border-formal-accent'
+    ),
+    outlineColor('focus-within:outline-primary'),
+    outlineStyle('focus-within:outline'),
+    borderRadius('rounded-lg'),
+    padding('py-3', 'px-4'),
+    transitionProperty('transition-colors'),
+    backgroundColor('bg-primary-dark')
+  )
+const textWithErrorWrapper = classnames(space('space-y-4'))
+
+const textBox = classnames(
+  backgroundColor('bg-primary-dark'),
+  resize('resize-none'),
+  width('w-full'),
+  transitionProperty('transition-colors'),
+  outlineStyle('outline-none', 'focus:outline-none')
+)
+
+interface TextAreaProps {
+  text: string
+  onTextChange: (text: string) => void
+  maxLength: number
+  disabled?: boolean
+  error?: unknown
+  footer?: string
+}
+
+export default function ({
+  text,
+  onTextChange,
+  disabled,
+  error,
+  maxLength,
+  ...restProps
+}: TextAreaProps & TextareaAutosizeProps) {
+  const isValid = !error && text.length <= maxLength
+
+  return (
+    <div className={textWithErrorWrapper}>
+      <div className={containerWithFooter}>
```

Comment:
Looks like this `div` and `div` from `L85` are doing nothing, except creating more nesting, can we combine them? Not sure if I am right here, but on `L101/L102` I can see 2 closing `</div>` tags, we don't have any special effects (like rainbow outline), so I may be right about combining them

Reply:
I think you're right, fixed it. ty

Reply:
actually, it broke width of textinput. i've return it back